### DISCUSSION TITLE
JWT for access tokens

### DISCUSSION THREAD

**Discussion by sagarshah1983** [ory/hydra#2245] [Q&A] [answered]
I came across an article indicating how to configure jwt access tokens using just ory hydra.
Reference: https://www.ory.sh/hydra/docs/advanced/#json-web-tokens

For now, I am running hydra using docker container, and not very certain where does this configuration fall into?

```
strategies:
  access_token: jwt
```

Can someone please clarify?

Any help is highly appreciated!

**Comment by k9ert** [comment, marked as answer]
Please have a look into the hydra-reference-documentation/configuration here:
https://www.ory.sh/hydra/docs/reference/configuration

So in the case of the 5 minute tutorial, this configuration is located in `contrib/quickstart/5-min/hydra.yml`.
Does this help you?

**Reply by sagarshah1983** [reply]
Thank you for responding!
I am trying to use docker container as follows. if you could clarify, how do I pass the parameter to my container, that would be helpful.

`docker run -d --name ory-hydra-example--hydra --network hydraguide 
  -p 9000:4444 -p 9001:4445 -e SECRETS_SYSTEM=$SECRETS_SYSTEM 
  -e DSN=$DSN -e URLS_SELF_ISSUER=http://127.0.0.1:9000/ -e URLS_CONSENT=http://127.0.0.1:8080/consent.html 
  -e URLS_LOGIN=http://127.0.0.1:8080/login.html 
  -e TTL_REFRESH_TOKEN=2190h 
  oryd/hydra:v1.7.4 serve all --dangerous-force-http`

Appreciate!

**Reply by vinckr** [reply]
This seems more like a question on docker usage, I am sure there are plenty of [resources](https://docs.docker.com/compose/environment-variables/) out there that can help you here. 
But essentially you want to include the right hydra.yml in your container. 

As for using the jwt method with Ory Hydra, there is a portion in the doc you linked, recommending Ory Oathkeeper if you want to use the jwt method:

> There is a better solution: Use ORY Oathkeeper! ORY Oathkeeper is a proxy you deploy in front of your services. It will "convert" ORY Hydra's opaque Access Tokens into JSON Web Tokens for your backend services. This allows your services to work without additional REST Calls while solving all previous points. We really recommend this option if you want JWTs!

**Reply by sagarshah1983** [reply]
Thanks for clarifications. I have not explored yet on oathkeeper, as we plan to use API gateway anyway in front. But will definitely take a look at it. Appreciate your recommendations.

**Reply by sagarshah1983** [reply]
Thank you! I was able to make it work with the additional env parameter (which I was missing earlier but later found in quickstart-jwt.yml from codebase) file from codebase.
`-e STRATEGIES_ACCESS_TOKEN=jwt`
